### DISCUSSION TITLE
Grid layout - With header/footer and growing nav/main/aside sections ('holy grail' layout)

### DISCUSSION THREAD

**Discussion by MartijnHarmenzon** [tailwindlabs/tailwindcss#1777] [Help]
I am trying to create a dashboard with the new Tailwind Grid functionality. In the end it would look like the image below. Where the nav, main and aside will grow to a full height (h-full). The header and footer will have a max-height.

image
![image](https://user-images.githubusercontent.com/23123158/82048676-5a82f400-96b5-11ea-9859-51d2d032c0d7.png)

The code I am using at the moment is the following:
```
<div class="grid grid-rows-3 grid-cols-12 gap-2 h-screen bg-gray-100">
  <header class="row-auto col-span-12 bg-gray-300">
    <!-- Header content -->
    Header...
  </header>

  <nav class="col-span-1 bg-gray-400">
    <!-- Navigation -->
    Nav...
  </nav>

  <main class="col-span-8 bg-gray-500">
    <!-- Main content -->
    Main...
  </main>

  <aside class="col-span-3 bg-gray-600">
    <!-- Sidebar / Ads -->
    Aside...
  </aside>

  <footer class="row-auto col-span-12 bg-gray-300">
    <!-- Footer content -->
    Footer...
  </footer>
</div>
```

And a Codepen:
https://codepen.io/martijnharmenzon-the-looper/pen/vYOEGRQ

Any suggestions how to make the nav, main and aside grow? Much appreciated.

**Comment by estevanmaito** [comment]
Change your grid from `grid-rows-3` to  `grid-rows-6` and then add to `aside`, `nav `and `main ` the class `row-span-4`.

It will expand the rows of your grid and make those 3 items fill the space.

**Comment by browner12** [comment]
Would this be a better case for flex, at least on your cross axis?  I do this exact layout on a lot of my sites.

```html
<div id="app" class="flex flex-col h-screen w-screen">
    <div id="header"></div>
    <div id="main" class="grid grid-cols-3 flex-grow"></div>
    <div id="footer"></div>
</div>
```

**Comment by Postroggy** [comment]
yeah, that should work perfectly. but I tried with grid but I am not using tailwindcss, here is my code : 
```vue
<template>
  <div class="full-content">
    <header class="text-pink-400">header</header>
    <main>
      <div v-for="item in items" :key="item">{{ item.text }}</div>
    </main>
    <footer>Footer</footer>
  </div>
</template>

<script>
import { ref } from "vue";
import "./home.less";

export default {
  setup() {
    const msg = ref("Home Pages");
    const items = ref([{ text: "Hello1" }, { text: "Hello2" }]);
    return {
      msg,
      items,
    };
  },
};
</script>
<style src="./home.less">
</style>
```
and my style.css
```less
.full-content {
  height: 100vh;
  display: grid;
  grid-template-rows: auto 1fr auto;
  grid-template-columns: 1fr;

  header {
      color: red;
  }
  main{
    overflow: auto;
  }

  footer {
    grid-row-start: 12;
    color: red;
  }
}

```
that just worked fine as `flex flex-col h-screen w-screen` . 
**But I don't know how to set `grid-template-rows` to `auto 1fr auto` with Tailwindcss Functionlity**
